src/runner.rs

```diff
@@ -102,8 +102,25 @@ pub fn run(av: ArgValues) -> error::Result<()> {
     });
 
     let event_loop = EventLoop::new();
+
+    let shader_name = av
+        .getid
+        .as_ref()
+        .or_else(|| av.shaderpath.as_ref())
```

Comment:
I just realized, I think these can change to `.or(...)` to get rid of the closure.